sql/tests: nil pointer in test builds in forwardInnerQueryStats

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21003005?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21003005?buildTab=artifacts#/) on release-26.1 @ [6ac107383aa8b7246f0d6448366407b6f53e71e4](https://github.com/cockroachdb/cockroach/commits/6ac107383aa8b7246f0d6448366407b6f53e71e4):


```
    rsg_test.go:804: CREATE TABLE "t-able_1" ("\vcol1_0" LTREE NOT NULL, "col1�_1" REGCLASS NULL, "co\\x17l1_2" PG_LSN, "col1%v_3" BOX2D, "\\U0005DDF1col1_4" PG_LSN NOT NULL, "col1 _5" UUID NOT NULL, col1_6 FLOAT8, "\\xadc\\x03ol1_7" NAME NOT NULL, col1_8 JSONB, "co%ql1_9" "char", col1_10 STRING AS (lower(CAST("co\\x17l1_2" AS STRING))) STORED, col1_11 STRING NOT NULL AS (lower("co%ql1_9")) STORED, col1_12 FLOAT8 NOT NULL AS (abs(col1_6)) VIRTUAL, col1_13 STRING NOT NULL AS (lower(CAST("\\U0005DDF1col1_4" AS STRING))) STORED, " col1_14" STRING NOT NULL AS (lower(CAST("col1 _5" AS STRING))) VIRTUAL, PRIMARY KEY (col1_12 DESC, col1_11 DESC), INVERTED INDEX (col1_11 DESC, col1_10 DESC, "\\xadc\\x03ol1_7" gin_trgm_ops ASC), INDEX ("co%ql1_9" ASC, lower(CAST("co\\x17l1_2" AS STRING)) ASC, "\vcol1_0" ASC, "col1 _5" DESC) STORING ("col1�_1", "co\\x17l1_2", "col1%v_3", "\\U0005DDF1col1_4", col1_6, col1_13) WHERE (((((("t-able_1".col1_11 > e'\x00':::STRING) OR ("t-able_1"." col1_14" > e'\x00':::STRING)) AND ("t-able_1".col1_13 < 'X':::STRING)) AND ("t-able_1"."co%ql1_9" = e'\x00':::STRING)) AND ("t-able_1".col1_6 < (-0.0):::FLOAT8)) OR ("t-able_1"."\\xadc\\x03ol1_7" > e'\'':::STRING:::NAME)) AND ("t-able_1".col1_10 >= e'\'':::STRING), UNIQUE (col1_12, abs(col1_6), "\\xadc\\x03ol1_7" ASC, col1_11, "\vcol1_0" ASC, col1_8, col1_10, "co%ql1_9", lower("\\xadc\\x03ol1_7"), "col1�_1"), UNIQUE ("co\\x17l1_2" ASC, col1_8 ASC) PARTITION BY LIST ("co\\x17l1_2") (PARTITION "t-able_1_part_0" VALUES IN (('BFB9D35F/53E93F85':::PG_LSN,), ('4FA734AF/6F74B279':::PG_LSN,), ('864BEDD6/1715671':::PG_LSN,), ('7FFFFFFF/FFFFFFFF':::PG_LSN,), ('FFFFFFFF/FFFFFFFF':::PG_LSN,), ('F314FF5A/F47C6785':::PG_LSN,), ('639A7094/602F0A51':::PG_LSN,)), PARTITION "t-able_1_part_1" VALUES IN (('B9EBFE70/FA756FE9':::PG_LSN,), ('CE7960DE/192E9B0A':::PG_LSN,), ('3B75F760/55B47F2C':::PG_LSN,), ('3263573C/A1F62008':::PG_LSN,), ('EFAAB4D3/B9299FF5':::PG_LSN,), ('A2549D60/3D7E9DCD':::PG_LSN,), ('72C23DD7/37F6A355':::PG_LSN,)), PARTITION "t-a͊ble_1_part_2" VALUES IN (('7BC1DD57/69AEB0FB':::PG_LSN,), ('64ACF4DB/25608477':::PG_LSN,), ('592AC63F/10EE82C3':::PG_LSN,), ('B6BF8711/57589E86':::PG_LSN,), ('CC82D1D7/45AD74B8':::PG_LSN,), ('D6BF8E53/DE23E46C':::PG_LSN,), ('1335B8E8/4BE19CE6':::PG_LSN,)), PARTITION "t-able_1_p😛art_3" VALUES IN (('A55B6D82/6C3DEBF4':::PG_LSN,), ('37437A2/F49B539':::PG_LSN,), ('BCCA846F/75786D1A':::PG_LSN,), ('5252D7C9/C211E58C':::PG_LSN,), ('D3D87AE3/5D3B1609':::PG_LSN,), ('DAAA66FF/90C1BB18':::PG_LSN,), ('B042504C/564D55D9':::PG_LSN,)), PARTITION "t-able_1_part_4" VALUES IN (('6F81FEFA/A10DA04D':::PG_LSN,), ('AA381AE8/7A1D2567':::PG_LSN,), ('7BC144F2/A5C14FF2':::PG_LSN,), ('5DD06F33/512A8F2A':::PG_LSN,), ('6AC49EE1/48ACF97A':::PG_LSN,), ('E8F2B54B/81428354':::PG_LSN,), ('7FABA6AE/6D9EA320':::PG_LSN,)), PARTITION "t-able_1_part_5" VALUES IN (('59ECFCCF/F338267':::PG_LSN,), ('27566650/4071005A':::PG_LSN,), ('538BC33F/1961FBD6':::PG_LSN,), ('3794FF1A/F9F8C07F':::PG_LSN,), ('F6FD9957/B1C490D6':::PG_LSN,), ('58FF3DAB/89EF64E3':::PG_LSN,), ('0/0':::PG_LSN,))) WHERE ((((((("t-able_1"." col1_14" = '"':::STRING) OR ("t-able_1".col1_6 >= '-Inf':::FLOAT8)) OR ("t-able_1"."co%ql1_9" < e'\'':::STRING)) OR ("t-able_1".col1_11 = e'\x00':::STRING)) OR ("t-able_1".col1_13 = e'\'':::STRING)) AND ("t-able_1"."\\xadc\\x03ol1_7" = '':::STRING:::NAME)) AND ("t-able_1".col1_12 >= '-Inf':::FLOAT8)) AND ("t-able_1".col1_10 != 'X':::STRING), INDEX (col1_10) STORING ("\\U0005DDF1col1_4", "col1 _5", col1_6, "co%ql1_9", col1_13) NOT VISIBLE, UNIQUE ("\\xadc\\x03ol1_7", "\vcol1_0", "co\\x17l1_2" ASC));
    rsg_test.go:804: ALTER TABLE "t-able_1" INJECT STATISTICS e'[{"avg_size": 25, "columns": ["\\"\\\\vcol1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2404461855043218528, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 6, "columns": ["\\"col1\\u000b_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3168423650799103536, "histo_col_type": "", "name": "__auto__", "null_count": 5495600023820777984, "row_count": 6854376877420892550}, {"avg_size": 13, "columns": ["\\"co\\\\\\\\x17l1_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6557267155721635840, "histo_buckets": [{"distinct_range": 0, "num_eq": 20000000, "num_range": 0, "upper_bound": "29AD0124/2BEFAECA"}, {"distinct_range": 0, "num_eq": 300000000, "num_range": 60000000, "upper_bound": "36C3390B/6306B40C"}, {"distinct_range": 782512030615473000, "num_eq": 4447035135760751616, "num_range": 1584195062608968960, "upper_bound": "10B86517/80D82766"}, {"distinct_range": 0, "num_eq": 0, "num_range": 7980043278188994560, "upper_bound": "329DB4F5/47D17EED"}, {"distinct_range": 4000000, "num_eq": 400000, "num_range": 4000000, "upper_bound": "8CC57EB/9A4D9A6E"}, {"distinct_range": 0, "num_eq": 1000, "num_range": 740164099060384000, "upper_bound": "1E5A4E4/C8C4177D"}], "histo_col_type": "PG_LSN", "histo_version": 3, "name": "__auto__", "null_count": 5563782452522830757, "row_count": 6854376877420892550}, {"avg_size": 18, "columns": ["\\"col1%v_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6306984533933080856, "histo_col_type": "", "name": "__auto__", "null_count": 6541252855280673489, "row_count": 6854376877420892550}, {"avg_size": 24, "columns": ["\\"\\\\\\\\U0005DDF1col1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1428671262390856352, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 27, "columns": ["\\"col1 _5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1525243924569065503, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 3, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4007331932525372043, "histo_col_type": "", "name": "__auto__", "null_count": 4000537693819568767, "row_count": 6854376877420892550}, {"avg_size": 18, "columns": ["\\"\\\\\\\\xadc\\\\\\\\x03ol1_7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6025677171209597952, "histo_buckets": [{"distinct_range": 0, "num_eq": 200, "num_range": 0, "upper_bound": ""}, {"distinct_range": 30000000000, "num_eq": 700000, "num_range": 30000000000, "upper_bound": "\\u0001"}, {"distinct_range": 0, "num_eq": 4000000000, "num_range": 8000000, "upper_bound": "\\u001f\\u000b"}, {"distinct_range": 924580003076816800, "num_eq": 500, "num_range": 5617460786801152000, "upper_bound": "v "}, {"distinct_range": 0, "num_eq": 4663767325947276288, "num_range": 4367393529175997952, "upper_bound": "wg\\u000f7"}, {"distinct_range": 8607266792463674000, "num_eq": 1481854698254542848, "num_range": 9180359038556625920, "upper_bound": "\\u00159\\u001a(%;"}, {"distinct_range": 1439446874019384300, "num_eq": 5041286883521219584, "num_range": 2432739724897308672, "upper_bound": ",\\u0006dA;i\x7f"}, {"distinct_range": 423896739896986400, "num_eq": 100000, "num_range": 423896739896986368, "upper_bound": "6\\u001aE``T\\t]B"}], "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 21, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3850474210234415122, "histo_col_type": "", "name": "__auto__", "null_count": 5655147194634488539, "row_count": 6854376877420892550}, {"avg_size": 30, "columns": ["\\"co%ql1_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4008998736552054784, "histo_buckets": [{"distinct_range": 0, "num_eq": 1551083885897633024, "num_range": 0, "upper_bound": " "}, {"distinct_range": 100, "num_eq": 2195235514950435328, "num_range": 100, "upper_bound": "\'"}, {"distinct_range": 900000000, "num_eq": 6875476602763251712, "num_range": 900000000, "upper_bound": "("}, {"distinct_range": 0, "num_eq": 2537798756902871552, "num_range": 0, "upper_bound": "/"}, {"distinct_range": 0, "num_eq": 60000, "num_range": 6000000000, "upper_bound": "9"}, {"distinct_range": 633867263828786000, "num_eq": 4000000000, "num_range": 633867263828786048, "upper_bound": "B"}, {"distinct_range": 3751676200823227400, "num_eq": 0, "num_range": 3751676200823227392, "upper_bound": "h"}], "histo_col_type": "\\"char\\"", "histo_version": 3, "name": "__auto__", "null_count": 4312322733636473287, "row_count": 6854376877420892550}, {"avg_size": 14, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 120526486875777744, "histo_buckets": [{"distinct_range": 0, "num_eq": 90000000, "num_range": 0, "upper_bound": "g"}, {"distinct_range": 12.997379727877714, "num_eq": 2019485137706966528, "num_range": 3000, "upper_bound": "\U00002603"}, {"distinct_range": 0, "num_eq": 3518542031909288448, "num_range": 50000, "upper_bound": "\'_\\u0013\\\\z"}, {"distinct_range": 4468327.185515279, "num_eq": 20000000000, "num_range": 9000000000, "upper_bound": "oD+tI"}, {"distinct_range": 6000000, "num_eq": 7971452130863059968, "num_range": 6000000, "upper_bound": "X5Fd``fX"}, {"distinct_range": 0, "num_eq": 70000, "num_range": 80, "upper_bound": "e\\u0017q\\\\9|\\u0001"}, {"distinct_range": 0, "num_eq": 1000000, "num_range": 0, "upper_bound": "Ef7$l\\u0019(B"}, {"distinct_range": 3387607963856356400, "num_eq": 8300020989077837824, "num_range": 3387607963856356352, "upper_bound": "``e.s\\u0002cBS"}, {"distinct_range": 300181639127330750, "num_eq": 5748160960571315200, "num_range": 4956331330746454016, "upper_bound": "cLk4uwj-"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 5660903118192467469, "row_count": 6854376877420892550}, {"avg_size": 26, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2925618431367028736, "histo_buckets": [{"distinct_range": 0, "num_eq": 5802455265420340224, "num_range": 0, "upper_bound": ""}, {"distinct_range": 613931613797932300, "num_eq": 20000000000, "num_range": 5619291527514868736, "upper_bound": "y&"}, {"distinct_range": 100000000000, "num_eq": 2514584604494984192, "num_range": 100000000000, "upper_bound": "B@%"}, {"distinct_range": 0, "num_eq": 600, "num_range": 20, "upper_bound": "9\\u000b9\\u001e#"}, {"distinct_range": 0, "num_eq": 8000000, "num_range": 6532572851155298304, "upper_bound": "``\\n}n\\u0016"}, {"distinct_range": 7000, "num_eq": 6000000, "num_range": 7000, "upper_bound": "7\\u001az\\u001e\\u000c\\n\\u0013"}, {"distinct_range": 0, "num_eq": 200, "num_range": 5249805111236454400, "upper_bound": "\x7f(VE;3\\u0007"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 2, "columns": ["col1_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1973748043538353664, "histo_buckets": [{"distinct_range": 0, "num_eq": 80000000000, "num_range": 0, "upper_bound": "0.4037464760388998"}, {"distinct_range": 0, "num_eq": 2570602583553595392, "num_range": 8000000000, "upper_bound": "1.0545531324396265"}, {"distinct_range": 80000, "num_eq": 1274811924722041856, "num_range": 80000, "upper_bound": "2.3179709809007685"}, {"distinct_range": 0, "num_eq": 30000000000, "num_range": 339102124176654336, "upper_bound": "-0.0"}, {"distinct_range": 4740377896771092000, "num_eq": 4000, "num_range": 6522957066691453952, "upper_bound": "-0.50482245493836"}, {"distinct_range": 2676794334277182000, "num_eq": 1177001656279065600, "num_range": 5903120552762478592, "upper_bound": "-0.5868153439697311"}, {"distinct_range": 0, "num_eq": 1000, "num_range": 6819028887919239168, "upper_bound": "-2.1885281190170494"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 7, "columns": ["col1_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6486861804194641479, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}, {"avg_size": 23, "columns": ["\\" col1_14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3980640695701760537, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6854376877420892550}]':::JSONB;
    rsg_test.go:804: CREATE TYPE rand_typ_0 AS ENUM ('lyxaqz', 'sx', 'tlj');
    rsg_test.go:804: CREATE TYPE rand_typ_1 AS ENUM ('gr', 'lpovy', 'xhiami', 'xrvow');
    rsg_test.go:804: CREATE TYPE rand_typ_2 AS (b TIMESTAMP, n LTREE, bbt LTREE);
    rsg_test.go:804: CREATE TYPE rand_typ_3 AS ENUM ('llnjka', 'eq', 'pjwu', 'uefho');
    rsg_test.go:804: CREATE TYPE rand_typ_4 AS ENUM ('sars', 'p', 'se', 'rrdin', 'ceqcjg', 'lk');
    rsg_test.go:804: SET sql_safe_updates = false;;
    rsg_test.go:804: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (8330318773649783673);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (728361459495643440);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (2846639916107662573);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (1537596874033748004);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (549817505756152114);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (539112651000083368);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (6358033172754125244);
    rsg_test.go:804: ALTER TABLE seed SCATTER;
    rsg_test.go:806: 
    rsg_test.go:807: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1801851922
--- FAIL: TestRandomSyntaxSQLSmith (302.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58895

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21052558?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21052558?buildTab=artifacts#/) on release-26.1 @ [ce8399445d85ee35a38fb84fe482aef882ab6dd0](https://github.com/cockroachdb/cockroach/commits/ce8399445d85ee35a38fb84fe482aef882ab6dd0):


```
    rsg_test.go:804: ALTER TABLE "%pta,ble_3" INJECT STATISTICS e'[{"avg_size": 24, "columns": ["co\u0342l3_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1387533, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 14, "columns": ["col3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2332115, "histo_col_type": "", "name": "__auto__", "null_count": 1746030, "row_count": 5000000}, {"avg_size": 21, "columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2779915, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 0, "columns": ["col3_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1249985, "histo_col_type": "", "name": "__auto__", "null_count": 233141, "row_count": 5000000}, {"avg_size": 6, "columns": ["col3_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4573800, "histo_col_type": "", "name": "__auto__", "null_count": 2661814, "row_count": 5000000}, {"avg_size": 15, "columns": ["col3_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2146173, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 30, "columns": ["col3_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4598179, "histo_col_type": "", "name": "__auto__", "null_count": 57452, "row_count": 5000000}, {"avg_size": 30, "columns": ["col3_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 925281, "histo_buckets": [{"distinct_range": 0, "num_eq": 50000, "num_range": 0, "upper_bound": "ARRAY[]"}, {"distinct_range": 0, "num_eq": 1535660176583886336, "num_range": 500, "upper_bound": "ARRAY[\'00543f62-3b33-42f9-bf9b-e44d8678f71c\':::UUID]"}, {"distinct_range": 0, "num_eq": 7793985863785621504, "num_range": 4596068553940314624, "upper_bound": "ARRAY[\'d887457c-fa80-4798-b142-ae5513fa6ea9\':::UUID]"}, {"distinct_range": 3718003211.202277, "num_eq": 70000000000, "num_range": 4000000000, "upper_bound": "ARRAY[\'40ea43b1-32b9-40af-bfa7-96380dd732e6\':::UUID,\'b2e9c51c-f14b-417a-99ef-16c5f1b206b1\':::UUID]"}, {"distinct_range": 0, "num_eq": 10000000, "num_range": 745128475725704064, "upper_bound": "ARRAY[\'54cbdf83-af2f-4685-8179-5917debf581c\':::UUID,\'10fab1bf-5b6f-4bb2-89ab-5bfd65caca33\':::UUID]"}, {"distinct_range": 3097314641175375000, "num_eq": 4316771279092126208, "num_range": 4246924426897164800, "upper_bound": "ARRAY[\'6e481657-4b19-4aae-a177-cce0d7620357\':::UUID,\'ffffffff-ffff-ffff-ffff-ffffffffffff\':::UUID,\'cd33d243-d5b8-4f55-8de8-23938fb4b728\':::UUID,\'6dec5525-8806-4f49-8f8c-c07a5903d161\':::UUID]"}], "histo_col_type": "UUID[]", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 15, "columns": ["col3_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4151436, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 9, "columns": ["col3_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2029244, "histo_col_type": "", "name": "__auto__", "null_count": 2671536, "row_count": 5000000}, {"avg_size": 2, "columns": ["col3_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1227550, "histo_col_type": "", "name": "__auto__", "null_count": 3562850, "row_count": 5000000}, {"avg_size": 28, "columns": ["col3_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2532015, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 18, "columns": ["col3_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4947836, "histo_col_type": "", "name": "__auto__", "null_count": 826553, "row_count": 5000000}, {"avg_size": 13, "columns": ["\\"col3 _13\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3070387, "histo_col_type": "", "name": "__auto__", "null_count": 768540, "row_count": 5000000}, {"avg_size": 15, "columns": ["\\"c%poL3\'_14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2181536, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 26, "columns": ["\\"col3\\\\r_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4964830, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000000}, {"avg_size": 0, "columns": ["col3_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4007575, "histo_col_type": "", "name": "__auto__", "null_count": 1918444, "row_count": 5000000}]':::JSONB;
    rsg_test.go:804: ALTER TABLE table_4 INJECT STATISTICS e'[{"avg_size": 31, "columns": ["\\"col{4_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 272343415, "histo_buckets": [{"distinct_range": 0, "num_eq": 100, "num_range": 0, "upper_bound": "[\\"k7pA\\", 0.11791907453768935, null, false]"}, {"distinct_range": 7720064918691159000, "num_eq": 800000, "num_range": 9080222752160670720, "upper_bound": "[null, {\\"EWt7\\": 0.7577790937607244, \\"id9hTFxbm\\": false}, true, true]"}, {"distinct_range": 974218743537803500, "num_eq": 80000000, "num_range": 1255761263367642880, "upper_bound": "{\\"baz\\": [{\\"Ebc\\": \\"foo\\", \\"sZATP0i0\\": \\"jZojTkQv\\"}, false], \\"gE\\": true, \\"rOd4Ltmje\\": true, \\"xR2Fqjwf\\": {\\"b\\": [null], \\"bar\\": {\\"9gKR8\\": false, \\"foo\\": true, \\"foobar\\": [false]}, \\"foo\\": true}}"}, {"distinct_range": 0, "num_eq": 6044256804096197632, "num_range": 80, "upper_bound": "{\\"a\\": {\\"VSoM\\": true, \\"XOr83YUlt\\": null}, \\"acEzTMy\\": 0.32445272630339805, \\"foobar\\": {\\"baz\\": 1.0905567723599785, \\"foobar\\": [null, true], \\"wV\\": {\\"1ryp\\": {\\"Tg\\": true, \\"a\\": \\"RSNpnWN\\", \\"b\\": true, \\"gbBY\\": {\\"bar\\": true, \\"baz\\": true}}, \\"Y3z4GIt\\": \\"bar\\", \\"b\\": [0.8717566844384455, false, null], \\"bar\\": false}}, \\"iIMI\\": 1.5003177624211603, \\"jaOa\\": false}"}], "histo_col_type": "JSONB", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 800000000}]':::JSONB;
    rsg_test.go:804: ALTER TABLE table_5 INJECT STATISTICS e'[{"avg_size": 8, "columns": ["\\"col5*_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 52723074447299801, "histo_col_type": "", "name": "__auto__", "null_count": 498836594995276538, "row_count": 3425014968496173016}, {"avg_size": 17, "columns": ["c\U0001F610ol5_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 914316538018484544, "histo_col_type": "", "name": "__auto__", "null_count": 1769598997695294953, "row_count": 3425014968496173016}, {"avg_size": 25, "columns": ["\\"col\\\\f5_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3152984105733704733, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 22, "columns": ["\\"co\\\\fl5\U0001F635_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1218946049993609843, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 15, "columns": ["c\U0001F60Dol\U0001F6205_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2700468421135795423, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 22, "columns": ["\\"col5\\n_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1213702618889690368, "histo_buckets": [{"distinct_range": 0, "num_eq": 1000000000, "num_range": 0, "upper_bound": "00000000-0000-0000-0000-000000000000"}, {"distinct_range": 0, "num_eq": 3000, "num_range": 5778040600655746048, "upper_bound": "15dfac29-1589-441a-9131-eae846f48fb3"}, {"distinct_range": 0, "num_eq": 10000, "num_range": 4904843929347052544, "upper_bound": "44fd1125-c1db-4a88-b978-798da8f267c6"}, {"distinct_range": 2407024.444916872, "num_eq": 200000, "num_range": 3000000, "upper_bound": "4a7763bf-0dbc-42c9-ace3-ea4caa875d60"}, {"distinct_range": 4713014243330648000, "num_eq": 513138090976484224, "num_range": 7394001248693605376, "upper_bound": "e8813902-05a3-47bb-b42f-90bcc7b6bfa2"}, {"distinct_range": 739677878780365600, "num_eq": 9000000, "num_range": 3246826619054214656, "upper_bound": "ef3587e0-6278-4942-be1e-e5fb19fcf1b3"}], "histo_col_type": "UUID", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 25, "columns": ["col5_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2351774875044861952, "histo_buckets": [{"distinct_range": 0, "num_eq": 800, "num_range": 0, "upper_bound": "00:00:00+15:59"}, {"distinct_range": 876236100184327300, "num_eq": 6465282488267473920, "num_range": 876236100184327296, "upper_bound": "08:15:24.697421-04:43"}, {"distinct_range": 10000000000, "num_eq": 0, "num_range": 10000000000, "upper_bound": "23:59:59.222948+08:28"}], "histo_col_type": "TIMETZ", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 19, "columns": ["\\"coL5_7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2703700993797580776, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 7, "columns": ["col5_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 930085450807551710, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 29, "columns": ["\\"coL5_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1586120653827720164, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 5, "columns": ["\\"co\\"\\"l5_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3158121245265315895, "histo_col_type": "", "name": "__auto__", "null_count": 353909704883347777, "row_count": 3425014968496173016}, {"avg_size": 29, "columns": ["col5_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3063424770246618646, "histo_col_type": "", "name": "__auto__", "null_count": 863062959271319559, "row_count": 3425014968496173016}, {"avg_size": 23, "columns": ["col5_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 703297487475270982, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 3, "columns": ["col5_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2229865030326969313, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}, {"avg_size": 22, "columns": ["\\"col%v5_14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2574592658642072971, "histo_col_type": "", "name": "__auto__", "null_count": 729931388171916222, "row_count": 3425014968496173016}, {"avg_size": 10, "columns": ["col5_15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2707526990656172142, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3425014968496173016}]':::JSONB;
    rsg_test.go:804: ALTER TABLE "t\fable_2" ADD FOREIGN KEY ("coL2_5") REFERENCES "%pta,ble_3" (col3_4) ON DELETE RESTRICT ON UPDATE RESTRICT;
    rsg_test.go:804: ALTER TABLE table_4 ADD FOREIGN KEY ("col{4_0") REFERENCES "%pta,ble_3" (col3_9) ON DELETE RESTRICT;
    rsg_test.go:804: CREATE TYPE rand_typ_0 AS (nay STRING, wqsol BIT(28), i REFCURSOR, g BOOL);
    rsg_test.go:804: CREATE TYPE rand_typ_1 AS (vg TIMESTAMP, ep INTERVAL, florr BOOL, owad GEOGRAPHY);
    rsg_test.go:804: CREATE TYPE rand_typ_2 AS (lyfw DATE, k UUID);
    rsg_test.go:804: CREATE TYPE rand_typ_3 AS ENUM ('lkx', 'jji', 'h', 'tixl', 'zu', 'qqb');
    rsg_test.go:804: SET sql_safe_updates = false;;
    rsg_test.go:804: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (7663970663232383200);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (5629169984388326493);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (5301011680495589839);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (7612559637686818108);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (6182367130201311522);
    rsg_test.go:804: ALTER TABLE seed SCATTER;
    rsg_test.go:806: 
    rsg_test.go:807: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith765660714
--- FAIL: TestRandomSyntaxSQLSmith (302.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/distsql_physical_planner.go
+++ b/pkg/sql/distsql_physical_planner.go
@@ -3575,6 +3575,14 @@ func (dsp *DistSQLPlanner) planJoiners(
 	return p
 }
 
+type noopMetadataForwarder struct{}
+
+var _ metadataForwarder = &noopMetadataForwarder{}
+
+func (n noopMetadataForwarder) forwardMetadata(*execinfrapb.ProducerMetadata) {}
+
+var singletonNoopMetadataForwarder = &noopMetadataForwarder{}
+
 // createPhysPlan creates a PhysicalPlan as well as returns a non-nil cleanup
 // function that must be called after the flow has been cleaned up.
 func (dsp *DistSQLPlanner) createPhysPlan(
@@ -3585,6 +3593,15 @@ func (dsp *DistSQLPlanner) createPhysPlan(
 		// plan.physPlan.onClose, so here we return a noop cleanup function.
 		return plan.physPlan.PhysicalPlan, func() {}, nil
 	}
+	// If we happen to evaluate an expression with a routine during the physical
+	// planning, we haven't yet set the metadata forwarder (we do that later,
+	// during the flow setup). So to prevent expected nil pointer internal
+	// errors we set a noop forwarder and then unset it once the physical
+	// planning is done.
+	planCtx.planner.routineMetadataForwarder = singletonNoopMetadataForwarder
+	defer func() {
+		planCtx.planner.routineMetadataForwarder = nil
+	}()
 	physPlan, err = dsp.createPhysPlanForPlanNode(ctx, planCtx, plan.planNode)
 	return physPlan, planCtx.getCleanupFunc(), err
 }

--- a/pkg/sql/logictest/testdata/logic_test/do
+++ b/pkg/sql/logictest/testdata/logic_test/do
@@ -326,3 +326,17 @@ BEGIN
   SELECT _float8, _int8, crdb_internal_mvcc_timestamp FROM seed@seed__int8__float8_idx;
 END;
 $$;
+
+# Regression test for hitting an internal error due to unset metadataForwarder
+# when a routine is evaluated during the physical planning (#161486).
+statement ok
+DO $$
+DECLARE
+BEGIN
+  SELECT sum(_float8) OVER (
+    PARTITION BY _int8 ROWS BETWEEN
+      UNBOUNDED PRECEDING AND
+      (SELECT _int8 FROM seed LIMIT 1) FOLLOWING
+  ) FROM seed;
+END;
+$$;

